File: qam-front/src/app/model/usuario.ts

import { DatePipe } from '@angular/common';

export class Usuario {
    
    userId:number;
    name:string;
    fechaModificacion:string;
    fechaRegistro:string;
    activo:boolean;
    user:string;
    password:string;
    ultimoLogin:string;
    ultimaActividad:string;
    build(content:any){
        const datepipe: DatePipe = new DatePipe('en-US')
        this.userId = content.userId;
        this.name = content.name;
        this.user = content.user;
     
        if(content.ultimoLogin!==null && content.ultimoLogin!==undefined){
        const ultimoL:any = datepipe.transform(content.ultimoLogin, 'dd-MM-YYYY HH:mm:ss');
        this.ultimoLogin = ultimoL!==null?ultimoL:"";
        }
        if(content.ultimaActividad!==null && content.ultimaActividad!==undefined){
        const ultimaA:any = datepipe.transform(content.ultimaActividad, 'dd-MM-YYYY HH:mm:ss');
        this.ultimaActividad = ultimaA!==null?ultimaA:"";
        }

        const fechaR:any = datepipe.transform(content.fechaRegistro, 'dd-MM-YYYY HH:mm:ss');
        this.fechaRegistro = fechaR!==null?fechaR:"";

        const fechaM:any = datepipe.transform(content.fechaModificacion, 'dd-MM-YYYY HH:mm:ss');
        this.fechaModificacion = fechaM!==null?fechaM:"";
        this.activo = content.activo;
    }
}

File: qam-ms/src/main/java/com/aura/admin/adminqamm/model/Usuario.java
package com.aura.admin.adminqamm.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;
import java.util.Set;

@Entity
@Table(name="usuario_nuevo")
@Getter
@Setter

public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id_usuario")
    private int userId;
    @Column(name = "display_name")
    private String name;
    @Column(name="fecha_modificacion")
    private Date fechaModificacion;
    @Column(name="fecha_registro")
    private Date fechaRegistro;
    @Column(name="ind_activo")
    private boolean activo;

    @Column(name="nombre_usuario")
    private String user;
    @Column(name="palabra_secreta")
    private String password;
    @Column(name="ultimo_login")
    private Date ultimoLogin;
    @Column(name="ultima_actividad")
    private Date ultimaActividad;

    @OneToMany(fetch = FetchType.LAZY,mappedBy = "usuario")
    @JsonBackReference(value="user-services")
    Set<Servicio> servicios;

    @OneToMany(fetch = FetchType.LAZY,mappedBy = "usuario")
    @JsonBackReference(value="user-interfaces")
    Set<Interfaz> interfaces;

    @OneToMany(fetch = FetchType.LAZY,mappedBy = "usuario")
    @JsonBackReference(value="user-userclients")
    Set<UsuarioCliente> usuarioClientes;

    @OneToMany(fetch = FetchType.LAZY,mappedBy = "usuario")
    @JsonBackReference(value="user-userroles")
    Set<UsuarioRol> roles;

    @OneToMany(fetch = FetchType.LAZY,mappedBy = "usuario")
    @JsonBackReference(value="user-userpermissions")
    Set<UsuarioPermiso> permisos;



}
